`apoc.import.csv` failed for NULL values when using `INT` datatype with error "[null] is not a supported property value"

## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use APOC, please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/neo4j-graph-platform/procedures-apoc) instead of creating an issue here.

## Expected Behavior (Mandatory)
CSV should get imported by ignoring the null value or may be a config to ignore null/empty  values can be added.

## Actual Behavior (Mandatory)
apoc.import.csv  fails with below error:
`
Failed to invoke procedure `apoc.import.csv`: Caused by: java.lang.IllegalArgumentException: [null] is not a supported property value
`

## How to Reproduce the Problem

The issue can be easily replicated using :

`CALL apoc.import.csv([{fileName: "file:/fail.csv", labels: ["entity"]}], [], {})`

Empty string value works fine in this case but error is thrown in case of integer property. I had a look at the documentation, the available config parameters does not contain any parameter to avoid nulls. 

The attachment pass.csv contains nulls in string and works fine.
fail.csv fails with above mentioned error as it contains null for Integer property.

### Simple Dataset (where it's possibile)

fail.csv
:ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT
n1,thing,once upon a time,1
n2,thing,there was a boy,2
n3,thing_variant,,3


pass.csv
:ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT
n1,thing,once upon a time,1
n2,thing,there was a boy,2
n3,thing_variant,,3


```
//Insert here a set of Cypher statements that helps us to reproduce the problem

CALL apoc.import.csv([{fileName: "file:/fail.csv", labels: ["entity"]}], [], {})


```


### Steps (Mandatory)

  1. Make sure compatible APOC version is installed  
  1. copy fail.csv and pass.csv in import folder 
  1.CALL apoc.import.csv([{fileName: "file:/fail.csv", labels: ["entity"]}], [], {})


## Screenshots (where it's possibile)

## Specifications (Mandatory)

String values having nulls got imported successfully . The issue exists with Integer property values. Available config parameters doesn't contain any parameter to skip nulls from properties.

Also interested to know if the string attribute can be interpreted as having no value instead of being a literal empty string?

Currently used versions

### Versions

  - OS:MacOS
  - Neo4j: 4.3.1
  - Neo4j-Apoc: 4.2.0.2

--- a/core/src/main/java/apoc/export/csv/CsvEntityLoader.java
+++ b/core/src/main/java/apoc/export/csv/CsvEntityLoader.java
@@ -136,7 +136,7 @@ public class CsvEntityLoader {
                         node.setProperty(field.getName(), idValue);
                         props++;
                     } else {
-                        boolean propertyAdded = CsvPropertyConverter.addPropertyToGraphEntity(node, field, value);
+                        boolean propertyAdded = CsvPropertyConverter.addPropertyToGraphEntity(node, field, value,clc);
                         props += propertyAdded ? 1 : 0;
                     }
                 }
@@ -232,7 +232,7 @@ public class CsvEntityLoader {
                 for (CsvHeaderField field : edgePropertiesFields) {
                     final String name = field.getName();
                     Object value = result.map.get(name);
-                    boolean propertyAdded = CsvPropertyConverter.addPropertyToGraphEntity(rel, field, value);
+                    boolean propertyAdded = CsvPropertyConverter.addPropertyToGraphEntity(rel, field, value, clc);
                     props += propertyAdded ? 1 : 0;
                 }
                 reporter.update(0, 1, props);

--- a/core/src/main/java/apoc/export/csv/CsvLoaderConfig.java
+++ b/core/src/main/java/apoc/export/csv/CsvLoaderConfig.java
@@ -15,6 +15,7 @@ public class CsvLoaderConfig {
     private static final String SKIP_LINES = "skipLines";
     private static final String BATCH_SIZE = "batchSize";
     private static final String IGNORE_DUPLICATE_NODES = "ignoreDuplicateNodes";
+    private static final String IGNORE_EMPTY_STRING = "ignoreEmptyString";
 
     private static char DELIMITER_DEFAULT = ',';
     private static char ARRAY_DELIMITER_DEFAULT = ';';
@@ -23,6 +24,7 @@ public class CsvLoaderConfig {
     private static int SKIP_LINES_DEFAULT = 1;
     private static int BATCH_SIZE_DEFAULT = 2000;
     private static boolean IGNORE_DUPLICATE_NODES_DEFAULT = false;
+    private static boolean IGNORE_EMPTY_STRING_DEFAULT = false;
 
     private final char delimiter;
     private final char arrayDelimiter;
@@ -31,6 +33,7 @@ public class CsvLoaderConfig {
     private final int skipLines;
     private final int batchSize;
     private final boolean ignoreDuplicateNodes;
+    private final boolean ignoreEmptyString;
 
     private CsvLoaderConfig(Builder builder) {
         this.delimiter = builder.delimiter;
@@ -40,6 +43,7 @@ public class CsvLoaderConfig {
         this.skipLines = builder.skipLines;
         this.batchSize = builder.batchSize;
         this.ignoreDuplicateNodes = builder.ignoreDuplicateNodes;
+        this.ignoreEmptyString = builder.ignoreEmptyString;
     }
 
     public char getDelimiter() {
@@ -67,6 +71,10 @@ public class CsvLoaderConfig {
     }
 
     public boolean getIgnoreDuplicateNodes() { return ignoreDuplicateNodes; }
+
+    public boolean isIgnoreEmptyString() {
+        return ignoreEmptyString;
+    }
 
     /**
      * Creates builder to build {@link CsvLoaderConfig}.
@@ -102,7 +110,7 @@ public class CsvLoaderConfig {
         if (config.get(SKIP_LINES) != null) builder.skipLines((int) config.get(SKIP_LINES));
         if (config.get(BATCH_SIZE) != null) builder.batchSize((int) config.get(BATCH_SIZE));
         if (config.get(IGNORE_DUPLICATE_NODES) != null) builder.ignoreDuplicateNodes((boolean) config.get(IGNORE_DUPLICATE_NODES));
-
+        if (config.get(IGNORE_EMPTY_STRING) != null) builder.ignoreEmptyString((boolean) config.get(IGNORE_EMPTY_STRING));
         return builder.build();
     }
 
@@ -117,6 +125,7 @@ public class CsvLoaderConfig {
         private int skipLines = SKIP_LINES_DEFAULT;
         private int batchSize = BATCH_SIZE_DEFAULT;
         private boolean ignoreDuplicateNodes = IGNORE_DUPLICATE_NODES_DEFAULT;
+        private boolean ignoreEmptyString = IGNORE_EMPTY_STRING_DEFAULT;
 
         private Builder() {
         }
@@ -153,6 +162,11 @@ public class CsvLoaderConfig {
 
         public Builder ignoreDuplicateNodes(boolean ignoreDuplicateNodes) {
             this.ignoreDuplicateNodes = ignoreDuplicateNodes;
+            return this;
+        }
+
+        public Builder ignoreEmptyString(boolean ignoreEmptyString) {
+            this.ignoreEmptyString = ignoreEmptyString;
             return this;
         }
 

--- a/core/src/main/java/apoc/export/csv/CsvLoaderConfig.java
+++ b/core/src/main/java/apoc/export/csv/CsvLoaderConfig.java
@@ -15,7 +15,7 @@ public class CsvLoaderConfig {
     private static final String SKIP_LINES = "skipLines";
     private static final String BATCH_SIZE = "batchSize";
     private static final String IGNORE_DUPLICATE_NODES = "ignoreDuplicateNodes";
-    private static final String IGNORE_EMPTY_STRING = "ignoreEmptyString";
+    private static final String IGNORE_BLANK_STRING = "ignoreBlankString";
 
     private static char DELIMITER_DEFAULT = ',';
     private static char ARRAY_DELIMITER_DEFAULT = ';';
@@ -24,7 +24,7 @@ public class CsvLoaderConfig {
     private static int SKIP_LINES_DEFAULT = 1;
     private static int BATCH_SIZE_DEFAULT = 2000;
     private static boolean IGNORE_DUPLICATE_NODES_DEFAULT = false;
-    private static boolean IGNORE_EMPTY_STRING_DEFAULT = false;
+    private static boolean IGNORE_BLANK_STRING_DEFAULT = false;
 
     private final char delimiter;
     private final char arrayDelimiter;
@@ -33,7 +33,7 @@ public class CsvLoaderConfig {
     private final int skipLines;
     private final int batchSize;
     private final boolean ignoreDuplicateNodes;
-    private final boolean ignoreEmptyString;
+    private final boolean ignoreBlankString;
 
     private CsvLoaderConfig(Builder builder) {
         this.delimiter = builder.delimiter;
@@ -43,7 +43,7 @@ public class CsvLoaderConfig {
         this.skipLines = builder.skipLines;
         this.batchSize = builder.batchSize;
         this.ignoreDuplicateNodes = builder.ignoreDuplicateNodes;
-        this.ignoreEmptyString = builder.ignoreEmptyString;
+        this.ignoreBlankString = builder.ignoreBlankString;
     }
 
     public char getDelimiter() {
@@ -72,8 +72,8 @@ public class CsvLoaderConfig {
 
     public boolean getIgnoreDuplicateNodes() { return ignoreDuplicateNodes; }
 
-    public boolean isIgnoreEmptyString() {
-        return ignoreEmptyString;
+    public boolean isIgnoreBlankString() {
+        return ignoreBlankString;
     }
 
     /**
@@ -110,7 +110,7 @@ public class CsvLoaderConfig {
         if (config.get(SKIP_LINES) != null) builder.skipLines((int) config.get(SKIP_LINES));
         if (config.get(BATCH_SIZE) != null) builder.batchSize((int) config.get(BATCH_SIZE));
         if (config.get(IGNORE_DUPLICATE_NODES) != null) builder.ignoreDuplicateNodes((boolean) config.get(IGNORE_DUPLICATE_NODES));
-        if (config.get(IGNORE_EMPTY_STRING) != null) builder.ignoreEmptyString((boolean) config.get(IGNORE_EMPTY_STRING));
+        if (config.get(IGNORE_BLANK_STRING) != null) builder.ignoreBlankString((boolean) config.get(IGNORE_BLANK_STRING));
         return builder.build();
     }
 
@@ -125,7 +125,7 @@ public class CsvLoaderConfig {
         private int skipLines = SKIP_LINES_DEFAULT;
         private int batchSize = BATCH_SIZE_DEFAULT;
         private boolean ignoreDuplicateNodes = IGNORE_DUPLICATE_NODES_DEFAULT;
-        private boolean ignoreEmptyString = IGNORE_EMPTY_STRING_DEFAULT;
+        private boolean ignoreBlankString = IGNORE_BLANK_STRING_DEFAULT;
 
         private Builder() {
         }
@@ -165,8 +165,8 @@ public class CsvLoaderConfig {
             return this;
         }
 
-        public Builder ignoreEmptyString(boolean ignoreEmptyString) {
-            this.ignoreEmptyString = ignoreEmptyString;
+        public Builder ignoreBlankString(boolean ignoreBlankString) {
+            this.ignoreBlankString = ignoreBlankString;
             return this;
         }
 

--- a/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
+++ b/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
@@ -22,7 +22,7 @@ public class CsvPropertyConverter {
             final Object[] array = list.toArray(prototype);
             entity.setProperty(field.getName(), array);
         } else {
-            if (config.isIgnoreEmptyString() && value instanceof String && StringUtils.isBlank((String) value)) {
+            if (config.isIgnoreBlankString() && value instanceof String && StringUtils.isBlank((String) value)) {
                 return false;
             }
             entity.setProperty(field.getName(), value);

--- a/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
+++ b/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
@@ -7,7 +7,7 @@ import java.util.List;
 public class CsvPropertyConverter {
 
     public static boolean addPropertyToGraphEntity(Entity entity, CsvHeaderField field, Object value) {
-        if (field.isIgnore()) {
+        if (field.isIgnore() || value == null) {
             return false;
         }
         if (field.isArray()) {

--- a/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
+++ b/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
@@ -3,6 +3,7 @@ package apoc.export.csv;
 import org.neo4j.graphdb.Entity;
 
 import java.util.List;
+import java.util.Objects;
 
 public class CsvPropertyConverter {
 
@@ -11,8 +12,13 @@ public class CsvPropertyConverter {
             return false;
         }
         if (field.isArray()) {
+            final List list = (List) value;
+            final boolean listContainingNull = list.stream().anyMatch(Objects::isNull);
+            if (listContainingNull) {
+                return false;
+            }
             final Object[] prototype = getPrototypeFor(field.getType());
-            final Object[] array = ((List<Object>) value).toArray(prototype);
+            final Object[] array = list.toArray(prototype);
             entity.setProperty(field.getName(), array);
         } else {
             entity.setProperty(field.getName(), value);
@@ -22,9 +28,9 @@ public class CsvPropertyConverter {
 
     static Object[] getPrototypeFor(String type) {
         switch (type) {
-            case "INT":     return new Integer  [] {};
+            case "INT":
             case "LONG":    return new Long     [] {};
-            case "FLOAT":   return new Float    [] {};
+            case "FLOAT":
             case "DOUBLE":  return new Double   [] {};
             case "BOOLEAN": return new Boolean  [] {};
             case "BYTE":    return new Byte     [] {};

--- a/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
+++ b/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
@@ -1,5 +1,6 @@
 package apoc.export.csv;
 
+import org.apache.commons.lang.StringUtils;
 import org.neo4j.graphdb.Entity;
 
 import java.util.List;
@@ -7,7 +8,7 @@ import java.util.Objects;
 
 public class CsvPropertyConverter {
 
-    public static boolean addPropertyToGraphEntity(Entity entity, CsvHeaderField field, Object value) {
+    public static boolean addPropertyToGraphEntity(Entity entity, CsvHeaderField field, Object value, CsvLoaderConfig config) {
         if (field.isIgnore() || value == null) {
             return false;
         }
@@ -21,6 +22,9 @@ public class CsvPropertyConverter {
             final Object[] array = list.toArray(prototype);
             entity.setProperty(field.getName(), array);
         } else {
+            if (config.isIgnoreEmptyString() && value instanceof String && StringUtils.isBlank((String) value)) {
+                return false;
+            }
             entity.setProperty(field.getName(), value);
         }
         return true;

--- a/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
+++ b/core/src/main/java/apoc/export/csv/CsvPropertyConverter.java
@@ -1,6 +1,6 @@
 package apoc.export.csv;
 
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.neo4j.graphdb.Entity;
 
 import java.util.List;

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -24,7 +24,10 @@ import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static apoc.util.MapUtil.map;
+import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertThat;
 
 public class ImportCsvTest {
@@ -103,10 +106,10 @@ public class ImportCsvTest {
                             "John\n" +
                             "Jane\n"),
                     new AbstractMap.SimpleEntry<>("emptyInteger", 
-                            ":ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT\n" +
-                            "n1,Thing,once upon a time,1\n" +
-                            "n2,Thing,there was a boy,2\n" +
-                            "n3,Thing,,\n"
+                            ":ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT,int_attribute_array:INT[],double_attribute_array:FLOAT[]\n" +
+                            "n1,Thing,once upon a time,1,\"2;3\",\"2.3;3.5\"\n" +
+                            "n2,Thing,there was a boy,2,\"4;5\",\"2.6;3.6\"\n" +
+                            "n3,Thing,,,,\n"
                     )
             ).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue)));
 
@@ -300,8 +303,22 @@ public class ImportCsvTest {
         TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {})",
                 r -> assertEquals(3L, r.get("nodes")));
 
-        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int ORDER BY n.int_attribute", 
-                r -> assertEquals(Arrays.asList(1L, 2L, null), Iterators.asList(r.columnAs("int"))));
+        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int, n.int_attribute_array as intArray, n.double_attribute_array as doubleArray ORDER BY n.int_attribute", 
+                r -> {
+                    final Map<String, Object> first = r.next();
+                    assertEquals(1L, first.get("int"));
+                    assertArrayEquals(new long[] { 2L, 3L }, (long[]) first.get("intArray"));
+                    assertArrayEquals(new double[] { 2.3D, 3.5D }, (double[]) first.get("doubleArray"), 0);
+                    final Map<String, Object> second = r.next();
+                    assertEquals(2L, second.get("int"));
+                    assertArrayEquals(new long[] { 4L, 5L }, (long[]) second.get("intArray"));
+                    assertArrayEquals(new double[] { 2.6D, 3.6D }, (double[]) second.get("doubleArray"), 0);
+                    final Map<String, Object> third = r.next();
+                    assertNull(third.get("int"));
+                    assertNull(third.get("intArray"));
+                    assertNull(third.get("doubleArray"));
+                    assertFalse(r.hasNext());
+                });
     }
 
     @Test

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -297,15 +297,11 @@ public class ImportCsvTest {
     
     @Test
     public void testEmptyInteger() {
-        TestUtil.testCall(db,
-                "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {})",
-                r -> assertEquals(3L, r.get("nodes"))
-        );
+        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {})",
+                r -> assertEquals(3L, r.get("nodes")));
 
-        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int ORDER BY n.int_attribute", r -> {
-            final List<Object> anInt = Iterators.asList(r.columnAs("int"));
-            assertEquals(Arrays.asList(1L, 2L, null), anInt);
-        });
+        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int ORDER BY n.int_attribute", 
+                r -> assertEquals(Arrays.asList(1L, 2L, null), Iterators.asList(r.columnAs("int"))));
     }
 
     @Test

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -8,6 +8,7 @@ import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.neo4j.configuration.GraphDatabaseSettings;
+import org.neo4j.graphdb.Node;
 import org.neo4j.graphdb.QueryExecutionException;
 import org.neo4j.internal.helpers.collection.Iterators;
 import org.neo4j.test.rule.DbmsRule;
@@ -108,7 +109,7 @@ public class ImportCsvTest {
                     new AbstractMap.SimpleEntry<>("emptyInteger", 
                             ":ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT,int_attribute_array:INT[],double_attribute_array:FLOAT[]\n" +
                             "n1,Thing,once upon a time,1,\"2;3\",\"2.3;3.5\"\n" +
-                            "n2,Thing,there was a boy,2,\"4;5\",\"2.6;3.6\"\n" +
+                            "n2,Thing,,2,\"4;5\",\"2.6;3.6\"\n" +
                             "n3,Thing,,,,\n"
                     )
             ).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue)));
@@ -300,23 +301,27 @@ public class ImportCsvTest {
     
     @Test
     public void testEmptyInteger() {
-        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {})",
+        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {ignoreEmptyString: true})",
                 r -> assertEquals(3L, r.get("nodes")));
 
-        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int, n.int_attribute_array as intArray, n.double_attribute_array as doubleArray ORDER BY n.int_attribute", 
+        TestUtil.testResult(db, "MATCH (node:Thing) RETURN node ORDER BY node.int_attribute",
                 r -> {
-                    final Map<String, Object> first = r.next();
-                    assertEquals(1L, first.get("int"));
-                    assertArrayEquals(new long[] { 2L, 3L }, (long[]) first.get("intArray"));
-                    assertArrayEquals(new double[] { 2.3D, 3.5D }, (double[]) first.get("doubleArray"), 0);
-                    final Map<String, Object> second = r.next();
-                    assertEquals(2L, second.get("int"));
-                    assertArrayEquals(new long[] { 4L, 5L }, (long[]) second.get("intArray"));
-                    assertArrayEquals(new double[] { 2.6D, 3.6D }, (double[]) second.get("doubleArray"), 0);
-                    final Map<String, Object> third = r.next();
-                    assertNull(third.get("int"));
-                    assertNull(third.get("intArray"));
-                    assertNull(third.get("doubleArray"));
+                    final Node firstNode = (Node) r.next().get("node");
+                    final Map<String, Object> firstProps = firstNode.getAllProperties();
+                    assertEquals(1L, firstProps.get("int_attribute"));
+                    assertArrayEquals(new long[] { 2L, 3L }, (long[]) firstProps.get("int_attribute_array"));
+                    assertArrayEquals(new double[] { 2.3D, 3.5D }, (double[]) firstProps.get("double_attribute_array"), 0);
+                    final Node secondNode = (Node) r.next().get("node");
+                    final Map<String, Object> secondProps = secondNode.getAllProperties();
+                    assertEquals(2L, secondProps.get("int_attribute"));
+                    assertArrayEquals(new long[] { 4L, 5L }, (long[]) secondProps.get("int_attribute_array"));
+                    assertArrayEquals(new double[] { 2.6D, 3.6D }, (double[]) secondProps.get("double_attribute_array"), 0);
+                    final Node thirdNode = (Node) r.next().get("node");
+                    final Map<String, Object> thirdProps = thirdNode.getAllProperties();
+                    assertNull(thirdProps.get("int_attribute"));
+                    assertNull(thirdProps.get("int_attribute_array"));
+                    assertNull(thirdProps.get("double_attribute_array"));
+                    assertNull(thirdProps.get("str_attribute"));
                     assertFalse(r.hasNext());
                 });
     }

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -9,12 +9,14 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.neo4j.configuration.GraphDatabaseSettings;
 import org.neo4j.graphdb.QueryExecutionException;
+import org.neo4j.internal.helpers.collection.Iterators;
 import org.neo4j.test.rule.DbmsRule;
 import org.neo4j.test.rule.ImpermanentDbmsRule;
 
 import java.io.File;
 import java.io.IOException;
 import java.util.AbstractMap;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
@@ -99,7 +101,13 @@ public class ImportCsvTest {
                             "2|Jane\n"),
                     new AbstractMap.SimpleEntry<>("personsWithoutIdField", "name:STRING\n" +
                             "John\n" +
-                            "Jane\n")
+                            "Jane\n"),
+                    new AbstractMap.SimpleEntry<>("emptyInteger", 
+                            ":ID(node_space_1),:LABEL,str_attribute:STRING,int_attribute:INT\n" +
+                            "n1,Thing,once upon a time,1\n" +
+                            "n2,Thing,there was a boy,2\n" +
+                            "n3,Thing,,\n"
+                    )
             ).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue)));
 
     @Before
@@ -285,6 +293,19 @@ public class ImportCsvTest {
 
         Assert.assertEquals("John Jane", TestUtil.singleResultFirstColumn(db, "MATCH (p1:Person)-[:FRIENDS_WITH]->(p2:Person) RETURN p1.name + ' ' + p2.name AS pair ORDER BY pair"));
         Assert.assertEquals("Jane John", TestUtil.singleResultFirstColumn(db, "MATCH (p1:Person)-[:KNOWS]->(p2:Person) RETURN p1.name + ' ' + p2.name AS pair ORDER BY pair"));
+    }
+    
+    @Test
+    public void testEmptyInteger() {
+        TestUtil.testCall(db,
+                "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {})",
+                r -> assertEquals(3L, r.get("nodes"))
+        );
+
+        TestUtil.testResult(db, "MATCH (n:Thing) RETURN n.int_attribute as int ORDER BY n.int_attribute", r -> {
+            final List<Object> anInt = Iterators.asList(r.columnAs("int"));
+            assertEquals(Arrays.asList(1L, 2L, null), anInt);
+        });
     }
 
     @Test

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -304,8 +304,7 @@ public class ImportCsvTest {
         TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {ignoreEmptyString: true})",
                 r -> assertEquals(3L, r.get("nodes")));
 
-        TestUtil.testResult(db, "MATCH (node:Thing) RETURN node ORDER BY node.int_attribute",
-                r -> {
+        TestUtil.testResult(db, "MATCH (node:Thing) RETURN node ORDER BY node.int_attribute", r -> {
                     final Node firstNode = (Node) r.next().get("node");
                     final Map<String, Object> firstProps = firstNode.getAllProperties();
                     assertEquals(1L, firstProps.get("int_attribute"));

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -301,7 +301,7 @@ public class ImportCsvTest {
     
     @Test
     public void testEmptyInteger() {
-        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {ignoreEmptyString: true})",
+        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyInteger.csv', labels: ['entity']}], [], {ignoreBlankString: true})",
                 r -> assertEquals(3L, r.get("nodes")));
 
         TestUtil.testResult(db, "MATCH (node:Thing) RETURN node ORDER BY node.int_attribute", r -> {

--- a/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
+++ b/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
@@ -10,5 +10,5 @@ The procedure support the following config parameters:
 | quotationCharacter | String | " | quotation character   | `--quote='"'`
 | stringIds | Boolean | true | treat ids as strings  | `--id-type=STRING`
 | skipLines | Integer | 1 | lines to skip (incl. header)  | N/A
-| ignoreEmptyString | Boolean | false | if true ignore properties with an empty string | N/A
+| ignoreBlankString | Boolean | false | if true ignore properties with a blank string | N/A
 |===

--- a/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
+++ b/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
@@ -10,4 +10,5 @@ The procedure support the following config parameters:
 | quotationCharacter | String | " | quotation character   | `--quote='"'`
 | stringIds | Boolean | true | treat ids as strings  | `--id-type=STRING`
 | skipLines | Integer | 1 | lines to skip (incl. header)  | N/A
+| ignoreEmptyString | Boolean | false | If true ignore properties with empty string | N/A
 |===

--- a/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
+++ b/docs/asciidoc/modules/ROOT/partials/usage/config/apoc.import.csv.adoc
@@ -10,5 +10,5 @@ The procedure support the following config parameters:
 | quotationCharacter | String | " | quotation character   | `--quote='"'`
 | stringIds | Boolean | true | treat ids as strings  | `--id-type=STRING`
 | skipLines | Integer | 1 | lines to skip (incl. header)  | N/A
-| ignoreEmptyString | Boolean | false | If true ignore properties with empty string | N/A
+| ignoreEmptyString | Boolean | false | if true ignore properties with an empty string | N/A
 |===

